ms2rescore/id_file_parser.py

```diff
@@ -178,11 +178,13 @@ def peprec_from_pin(self) -> PeptideRecord:
         peprec = self.original_pin.to_peptide_record(
             spectrum_index_pattern=self._pin_spec_id_patterns[self._pin_spec_id_style]
         )
-
-        # Map MGF titles and observed retention times
-        titles, retention_times = parse_mgf_title_rt(self.path_to_mgf_file)
-        peprec.df["observed_retention_time"] = peprec.df["spec_id"].map(retention_times)
-        peprec.df["spec_id"] = peprec.df["spec_id"].map(titles)
+        if "observed_retention_time" not in peprec.df.columns:
+            # Map MGF titles and observed retention times
+            titles, retention_times = parse_mgf_title_rt(self.path_to_mgf_file)
+            peprec.df["observed_retention_time"] = peprec.df["spec_id"].map(
+                retention_times
+            )
+            peprec.df["spec_id"] = peprec.df["spec_id"].map(titles)
```

Comment:
I had to implement use this because the retention times in the peprec file were more accurate then the ones from the mgf files (MHC quant output pin file). Depending on what we want as standard (RT times from MGF files or from peprec when they are parsed from the id file, we can leave it like this or change it around)